language/stdlib/modules/transaction_fee_distribution.mvir

```diff
@@ -0,0 +1,145 @@
+// This module implements a basic transaction fee distribution logic.
+//
+// We have made a couple design decisions here that are worth noting:
+//  * We pay out per-block as opposed to per-epoch. In the current distribution scheme (equal to
+//    all validators) there isn't any real difference between this and per-epoch distribution.
+//    However, other distribution schemes may have different economic implications depending upon
+//    if fees are distributed per-block or per-epoch.
+//  * In the handling of dust, it could be beneficial to be the first validator in the validating
+//    set. However, the maximal benefit that can be achieved per-block(/epoch) is at most 1
+//    microlibra * num_validators (which should be very very little).
+module TransactionFeeDistribution {
+  import 0x0.LibraCoin;
+  import 0x0.Block;
+  import 0x0.ValidatorSet;
+  import 0x0.LibraAccount;
+
+  resource T {
+      last_block_height_paid: u64,
+      fee_withdrawal_capability: LibraAccount.WithdrawalCapability,
+  }
+
+  // TODO: This is public for now for testing purposes. However, in the future block metadata
+  // transactions will call this and it will be marked as a private function.
+   public distribute_transaction_fees() acquires T {
+     let num_validators: u64;
+     let amount_to_distribute_per_validator: u64;
+     let amount_remainder: u64;
+     let lib_coin_to_distribute: LibraCoin.T;
+     let amount_collected: u64;
+     let distribution_resource: &mut Self.T;
+     let current_height: u64;
+
+     distribution_resource = borrow_global_mut<T>(0xFEE);
+     current_height = Block.get_current_height();
+
+     // TODO: This is removed for testing since block height remains constant.
+     //assert(*&copy(distribution_resource).last_block_height_paid < copy(current_height), 0);
+     // NB: We don't enforce sequentiality for transaction fee distribution.
```

Comment:
What is NB?

Reply:
equivalent to note well. I changed to something more understandable.